x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStatusAction.java

```diff
@@ -150,7 +158,30 @@ private GetStatusAction.Response getResponse(ClusterState state) {
             boolean dataStreamsPre891 = ProfilingDataStreamManager.isAnyResourceTooOld(state, indexStateResolver);
             boolean anyPre891Data = indicesPre891 || dataStreamsPre891;
 
-            return new GetStatusAction.Response(pluginEnabled, resourceManagementEnabled, resourcesCreated, anyPre891Data);
+            return new GetStatusAction.Response(pluginEnabled, resourceManagementEnabled, resourcesCreated, anyPre891Data, hasData(state));
+        }
+
+        private boolean hasData(ClusterState state) {
+            DataStream dataStream = state.metadata().dataStreams().get(EventsIndex.FULL_INDEX.getName());
+            if (dataStream == null) {
+                return false;
+            }
+            for (Index index : dataStream.getIndices()) {
+                IndexMetadata meta = state.metadata().index(index);
+                if (meta == null) {
+                    return false;
```

Comment:
What if there are many indices but only one doesn't have metadata? Should we not continue here?

Reply:
Fair point. I've pushed https://github.com/elastic/elasticsearch/pull/102735/commits/e11f4a8d7c3e76f0ff66dbd55a582cc72a9607ca.